### PR TITLE
fetchHenry: more plyr -> data.table conversions

### PR DESCRIPTION
**Pull request by brownag**

### CODE REVIEW

File: R/fetchHenry.R [hunk 1]
```diff
@@ -216,16 +161,27 @@ month2season <- function(x) {
     month = format(fake.datetimes, "%b")
   )
   
-  ## TODO: base::merge() is not a direct replacement, because it doesn't remove duplicate column names
-  # splice in missing data via full join
-  y <- join(x, fake.data, by = 'doy', type = 'full')
+  fill.cols <- which(!colnames(x) %in% colnames(fake.data))
```
Comment (dylanbeaudette) [inline R/fetchHenry.R:164]: Nice.

**Comment by dylanbeaudette** [comment]
Changes look good, thank you. I haven't tested. Running the fetchHenry tutorial might be a good test.

**Comment by brownag** [comment]
I used the fetchHenry tutorial dataset `project='CA792'` in my testing, it loads without error 

While working I added a couple tests for the builtin timeseries (`sid=469`) to ensure the NA padding works as expected (by checking resulting number of rows with and without padding)

**Comment by dylanbeaudette** [comment]
Perfect, thanks.

**Pull request merged**